packages/grafana-ui/src/components/Logs/ExploreGraphPanel.tsx

```diff
@@ -1,38 +1,62 @@
 import React, { PureComponent } from 'react';
-import { hot } from 'react-hot-loader';
-import { connect } from 'react-redux';
-import { LegendDisplayMode, GraphWithLegend } from '@grafana/ui';
-import { TimeZone, AbsoluteTimeRange, GraphSeriesXY, dateTimeForTimeZone, LoadingState } from '@grafana/data';
+import { css, cx } from 'emotion';
+import { TimeZone, AbsoluteTimeRange, GraphSeriesXY, dateTimeForTimeZone } from '@grafana/data';
 
-import { GraphSeriesToggler } from 'app/plugins/panel/graph2/GraphSeriesToggler';
-import Panel from './Panel';
-import { StoreState, ExploreId, ExploreMode } from 'app/types';
-import { getTimeZone } from '../profile/state/selectors';
-import { toggleGraph, updateTimeRange } from './state/actions';
+import { GraphSeriesToggler, GraphSeriesTogglerAPI } from '../Graph/GraphSeriesToggler';
+import { GraphWithLegend } from '../Graph/GraphWithLegend';
+import { LegendDisplayMode } from '../Legend/Legend';
+import { Panel } from '../Panel/Panel';
+import { Themeable, GrafanaTheme } from '../../types/theme';
+import { selectThemeVariant } from '../../themes/selectThemeVariant';
+import { withTheme } from '../../themes/ThemeContext';
 
 const MAX_NUMBER_OF_TIME_SERIES = 20;
 
-interface Props {
-  exploreId: ExploreId;
+const getStyles = (theme: GrafanaTheme) => ({
+  timeSeriesDisclaimer: css`
+    label: time-series-disclaimer;
+    width: 300px;
+    margin: ${theme.spacing.sm} auto;
+    padding: 10px 0;
+    border-radius: ${theme.border.radius.md};
+    text-align: center;
+    background-color: ${selectThemeVariant({ light: theme.colors.white, dark: theme.colors.dark4 }, theme.type)};
+  `,
+  disclaimerIcon: css`
+    label: disclaimer-icon;
+    color: ${theme.colors.yellow};
+    margin-right: ${theme.spacing.xs};
+  `,
+  showAllTimeSeries: css`
+    label: show-all-time-series;
+    cursor: pointer;
+    color: ${theme.colors.linkExternal};
+  `,
+});
+
+interface Props extends Themeable {
   series: GraphSeriesXY[];
   width: number;
-  absoluteRange?: AbsoluteTimeRange;
-  loading?: boolean;
-  mode?: ExploreMode;
-  showingGraph?: boolean;
-  showingTable?: boolean;
-  timeZone?: TimeZone;
+  absoluteRange: AbsoluteTimeRange;
+  loading: boolean;
+  showPanel: boolean;
+  showBars: boolean;
+  showLines: boolean;
+  isStacked: boolean;
+  showingGraph: boolean;
+  showingTable: boolean;
+  timeZone: TimeZone;
+  onUpdateTimeRange: (absoluteRange: AbsoluteTimeRange) => void;
+  onToggleGraph?: (showingGraph: boolean) => void;
   onHiddenSeriesChanged?: (hiddenSeries: string[]) => void;
-  toggleGraph: typeof toggleGraph;
-  updateTimeRange: typeof updateTimeRange;
 }
 
 interface State {
   hiddenSeries: string[];
   showAllTimeSeries: boolean;
 }
 
-export class ExploreGraphPanel extends PureComponent<Props, State> {
+class UnThemedExploreGraphPanel extends PureComponent<Props, State> {
```

Comment:
Minor consideration about this panel. In grafana-ui we keep components that are feature agnostic. I think this one, and any other that is Explore specific, should not belong to grafana-ui. Any thoughts on this @ryantxu ?

Reply:
Yes, I agree this does not look like it belongs here. Think Logs visualization should not use ExploreGraphPanel but the Graph component directly.